packages/integration-tests/tests/retrospective-demo/step4-discuss.test.ts

```diff
@@ -0,0 +1,178 @@
+import {test, expect} from '@playwright/test'
+import config from '../config'
+import {
+  startDemo,
+  goToNextPhase,
+  dragReflectionCard,
+  skipToDiscussPhase
+} from './retrospective-demo-helpers'
+
+test.describe('retrospective-demo / discuss page', () => {
+  test('it carries over user-created groups', async ({page, isMobile}) => {
+    await startDemo(page)
+
+    const startTextbox = '[data-cy=reflection-column-Start] [role=textbox]'
+    await page.click(startTextbox)
+    await page.type(startTextbox, 'Documenting things in Notion')
+    await page.press(startTextbox, 'Enter')
+
+    await page.click(startTextbox)
+    await page.type(startTextbox, 'Writing things down')
+    await page.press(startTextbox, 'Enter')
+
+    await goToNextPhase(page)
+    expect(page.url()).toEqual(`${config.rootUrlPath}/retrospective-demo/group`)
+
+    const writingThingsDownCard = page.locator('text=Writing things down')
+    const documentingInNotionCard = page.locator('text=Documenting things in Notion')
+    await dragReflectionCard(writingThingsDownCard, documentingInNotionCard)
+
+    // Then it auto-generates a header - we expect to see this in the sidebar on the discuss page
+    await expect(
+      page.locator(
+        `[data-cy=group-column-Start] [data-cy*="Start-group-"] input[value="Documenting things in"]`
+      )
+    ).toBeVisible()
+
+    await goToNextPhase(page)
+    expect(page.url()).toEqual(`${config.rootUrlPath}/retrospective-demo/vote`)
+    await goToNextPhase(page)
+    expect(page.url()).toEqual(`${config.rootUrlPath}/retrospective-demo/discuss/1`)
+
+    if (isMobile) {
+      await page.click('button[aria-label="Toggle the sidebar"]')
+    }
+
+    await expect(page.locator('[data-cy=sidebar] :text("Documenting things in")')).toBeVisible()
+  })
+
+  test('shows all the groups in the sidebar', async ({page, isMobile}) => {
+    await skipToDiscussPhase(page)
+
+    if (isMobile) {
+      await page.click('button[aria-label="Toggle the sidebar"]')
+    }
+
+    await expect(page.locator('[data-cy=sidebar] div:text("Meetings")')).toBeVisible()
+    await expect(page.locator('[data-cy=sidebar] div:text("Interns Staff")')).toBeVisible()
+    await expect(page.locator('[data-cy=sidebar] div:text("Work")')).toBeVisible()
+    await expect(page.locator('[data-cy=sidebar] div:text("Processes")')).toBeVisible()
+    await expect(page.locator('[data-cy=sidebar] div:text("Team")')).toBeVisible()
+    await expect(page.locator('[data-cy=sidebar] div:text("Debates")')).toBeVisible()
+    await expect(page.locator('[data-cy=sidebar] div:text("Decisions")')).toBeVisible()
+    await expect(page.locator('[data-cy=sidebar] div:text("People")')).toBeVisible()
+  })
+
+  interface DiscussTestCase {
+    name: string
+    emojis: Array<string>
+    comments?: Array<string>
+    tasks?: Array<string>
+  }
+
+  const discussTestCases: Array<DiscussTestCase> = [
```

Comment:
Here I'm auto-generating these tests to reduce duplication.

Reply:
-1 i'm getting some failures here. discussion have some randomness to em, perhaps that's the culprit? if so, i'd be happy to just test that someone can add a comment, maybe send a task to an integration, etc

results below:
```sh
  1) [chromium] › retrospective-demo/step4-discuss.test.ts:11:3 › retrospective-demo / discuss page › it carries over user-created groups 

    Error: expect(received).toBeVisible()
    Call log:
      - expect.toBeVisible with timeout 5000ms
      - waiting for selector "[data-cy=group-column-Start] [data-cy*="Start-group-"] input[value="Documenting things in"]"


      33 |         `[data-cy=group-column-Start] [data-cy*="Start-group-"] input[value="Documenting things in"]`
      34 |       )
    > 35 |     ).toBeVisible()
         |       ^
      36 |
      37 |     await goToNextPhase(page)
      38 |     expect(page.url()).toEqual(`${config.rootUrlPath}/retrospective-demo/vote`)

        at /Users/mattkrick/code/parabol/packages/integration-tests/tests/retrospective-demo/step4-discuss.test.ts:35:7

    attachment #1: trace (application/zip) ---------------------------------------------------------
    test-results/retrospective-demo-step4-discuss-retrospective-demo-discuss-page-it-carries-over-user-created-groups-chromium/trace.zip
    Usage:

        npx playwright show-trace test-results/retrospective-demo-step4-discuss-retrospective-demo-discuss-page-it-carries-over-user-created-groups-chromium/trace.zip

    ------------------------------------------------------------------------------------------------

    Retry #1 ---------------------------------------------------------------------------------------

    Error: expect(received).toBeVisible()
    Call log:
      - expect.toBeVisible with timeout 5000ms
      - waiting for selector "[data-cy=group-column-Start] [data-cy*="Start-group-"] input[value="Documenting things in"]"


      33 |         `[data-cy=group-column-Start] [data-cy*="Start-group-"] input[value="Documenting things in"]`
      34 |       )
    > 35 |     ).toBeVisible()
         |       ^
      36 |
      37 |     await goToNextPhase(page)
      38 |     expect(page.url()).toEqual(`${config.rootUrlPath}/retrospective-demo/vote`)

        at /Users/mattkrick/code/parabol/packages/integration-tests/tests/retrospective-demo/step4-discuss.test.ts:35:7

    attachment #1: trace (application/zip) ---------------------------------------------------------
    test-results/retrospective-demo-step4-discuss-retrospective-demo-discuss-page-it-carries-over-user-created-groups-chromium-retry1/trace.zip
    Usage:

        npx playwright show-trace test-results/retrospective-demo-step4-discuss-retrospective-demo-discuss-page-it-carries-over-user-created-groups-chromium-retry1/trace.zip

    ------------------------------------------------------------------------------------------------

    Retry #2 ---------------------------------------------------------------------------------------

    Error: expect(received).toBeVisible()
    Call log:
      - expect.toBeVisible with timeout 5000ms
      - waiting for selector "[data-cy=group-column-Start] [data-cy*="Start-group-"] input[value="Documenting things in"]"


      33 |         `[data-cy=group-column-Start] [data-cy*="Start-group-"] input[value="Documenting things in"]`
      34 |       )
    > 35 |     ).toBeVisible()
         |       ^
      36 |
      37 |     await goToNextPhase(page)
      38 |     expect(page.url()).toEqual(`${config.rootUrlPath}/retrospective-demo/vote`)

        at /Users/mattkrick/code/parabol/packages/integration-tests/tests/retrospective-demo/step4-discuss.test.ts:35:7

    attachment #1: trace (application/zip) ---------------------------------------------------------
    test-results/retrospective-demo-step4-discuss-retrospective-demo-discuss-page-it-carries-over-user-created-groups-chromium-retry2/trace.zip
    Usage:

        npx playwright show-trace test-results/retrospective-demo-step4-discuss-retrospective-demo-discuss-page-it-carries-over-user-created-groups-chromium-retry2/trace.zip
```

Reply:
Will you try again with my recent changes? I'm not getting any failures locally or on CI now.